nwb_conversion_tools/utils/conversion_tools.py

```diff
@@ -117,3 +65,43 @@ def estimate_recording_conversion_time(
 
     rmtree(temp_dir)
     return total_time, speed
+
+
+def run_conversion_from_yaml(file_path: FilePathType, overwrite: bool = False):
+    """
+    Run conversion to NWB given a yaml specification file.
+
+    Parameters
+    ----------
+    file_path : FilePathType
+        File path leading to .yml specification file for NWB conversion.
+    overwrite : bool, optional
+        If True, replaces any existing NWBFile at the nwbfile_path location, if save_to_file is True.
+        If False, appends the existing NWBFile at the nwbfile_path location, if save_to_file is True.
+        The default is False.
+
+    """
+    source_dir = Path(file_path).parent.absolute()
+    with open(file=file_path, mode="r") as io:
+        full_spec = yaml.load(stream=io, Loader=yaml.SafeLoader)
```

Comment:
I like separating things like this into a read function and an execute function that takes a dictionary as input. That allows a user to manually create a dictionary and use that as input instead of creating a YAML first

Reply:
So function 1 `yaml_to_dict` and then function 2 `run_conversion_from_yaml` that uses function 1?

Reply:
```python
def run_conversion_from_yaml(yaml):
    dict = yaml_to_dict(yaml)
    run_conversion_from_dict(dict)
```